src/datacrumbs/server/process/server.impl.cpp

```diff
@@ -0,0 +1,812 @@
+#endif
+#ifdef GET_DATA_3_EXISTS
+  DC_LOG_INFO("Collecting rest usdt events");
+  while (LOOKUP_3_CALL() != -1);
+#endif
+#ifdef GET_DATA_4_EXISTS
+  while (LOOKUP_4_CALL() != -1);
+#endif
+#ifdef GET_DATA_5_EXISTS
+  while (LOOKUP_5_CALL() != -1);
+#endif
+#ifdef GET_DATA_6_EXISTS
+  while (LOOKUP_6_CALL() != -1);
+#endif
+#ifdef GET_DATA_7_EXISTS
+  while (LOOKUP_7_CALL() != -1);
+#endif
+#ifdef GET_DATA_8_EXISTS
+  while (LOOKUP_8_CALL() != -1);
+#endif
+#ifdef GET_DATA_9_EXISTS
+  while (LOOKUP_9_CALL() != -1);
+#endif
+#endif
+#if defined(DATACRUMBS_MODE) && (DATACRUMBS_MODE == 1)
+  int failed_events = 0;
+  err = bpf_map_lookup_elem(failed_events_fd, &DATACRUMBS_FAILED_EVENTS_KEY, &failed_events);
+  if (err == 0) {
+    DC_LOG_PRINT("Total %d events failed", failed_events);
+  }
+#endif
+
+#if defined(DATACRUMBS_BPFTIME_COMPATIBLE_FLAG) && (DATACRUMBS_BPFTIME_COMPATIBLE_FLAG == 0)
+  DC_LOG_PRINT("Collecting string metadata from file_map...");
+  while (lookup_and_delete(file_hash_fd, event_processor, keys, values, batch_size, in_batch) ==
+         1) {
+    // Continue until no more keys are found
+  }
+  if (keys) {
+    free(keys);
+  }
+  if (values) {
+    free(values);
+  }
+#endif
+  DC_LOG_PRINT("Finalizing DataCrumbs...");
+  if (stop) {
+    DC_LOG_INFO("Received SIGINT (Ctrl-C), exiting gracefully");
+  }
+  // Measure elapsed time for finalization and cleanup
+  timer.resumeTime();
+
+  // Finalize ChromeWriter instance
+  event_processor->finalize();
+
+#if defined(DATACRUMBS_MODE) && (DATACRUMBS_MODE == 1)
+  // Cleanup resources
+  ring_buffer__free(rb);
+#endif
+  datacrumbs_bpf__destroy(skel);
+
+  double finalize_elapsed = timer.pauseTime();
+  DC_LOG_PRINT(
+      "Finalization and cleanup of DataCrumbs elapsed time: %f seconds with error code: %d",
+      finalize_elapsed, err);
+
+  // Write status info to JSON file for postprocessing
+
+  std::string status_file = "/tmp/datacrumbs_" + event_processor->configManager_->user +
+                            "_status_" + event_processor->configManager_->run_id + ".json";
+  struct json_object* status_json = json_object_new_object();
+  json_object_object_add(
+      status_json, "trace_file",
+      json_object_new_string(event_processor->configManager_->trace_file_path.c_str()));
+  json_object_object_add(status_json, "total_events_captured",
+                         json_object_new_int64(event_processor->event_index.load()));
+  json_object_object_add(status_json, "events_failed",
+                         json_object_new_int(event_processor->failed_events));
+
+  if (json_object_to_file_ext(status_file.c_str(), status_json, JSON_C_TO_STRING_PRETTY) != 0) {
+    DC_LOG_ERROR("Failed to write status file: %s", status_file.c_str());
+  }
+  json_object_put(status_json);
+  pwd = getpwnam(event_processor->configManager_->user.c_str());
+  uid = pwd ? pwd->pw_uid : static_cast<uid_t>(-1);
+  gid = pwd ? pwd->pw_gid : static_cast<gid_t>(-1);
```

Comment:
@trws, is the `-1` value correct here?

Reply:
It depends on what this is used for.  I'm pretty sure the "nobody" user on debian-style systems or any system with classic NFS users uses that as its default UID. If this is meant to be a "can't exist" or similar, it's not that.